doc/source/user/how-to-io.rst

```diff
@@ -0,0 +1,321 @@
+.. _how-to-io:
+
+##############################################################################
+Reading and writing files
+##############################################################################
+
+This page tackles common applications; for the full collection of I/O
+routines, see :ref:`routines.io`.
+
+
+******************************************************************************
+Reading text and CSV_ files
+******************************************************************************
+
+.. _CSV: https://en.wikipedia.org/wiki/Comma-separated_values
+
+With no missing values
+==============================================================================
+
+Use :func:`numpy.loadtxt`.
+
+With missing values
+==============================================================================
+
+Use :func:`numpy.genfromtxt`.
+
+:func:`numpy.genfromtxt` will either
+
+  - return a :ref:`masked array<maskedarray.generic>`
+    **masking out missing values** (if ``usemask=True``), or
+
+  - **fill in the missing value** with the value specified in
+    ``filling_values`` (default is ``np.nan`` for float, -1 for int).
+
+With non-whitespace delimiters
+------------------------------------------------------------------------------
+::
+
+    $ cat csv.txt  # doctest: +SKIP
+    1, 2, 3
+    4,, 6
+    7, 8, 9

```

Comment:
I think we've found ways to write this text to a temporary file as part of the refguide run, meaning we don't need to skip the doctests below. Do you remember where we did this @mattip? My hunch would be something like

```rst
.. refguide still finds this

    >>> with open('csv.txt', 'w') as f:
    ...    f.write("1, 2, 3\n4,, 6\n7, 8, 9\n")
```